Source: Written by Michael Forney
Upstream: Not interested (see DISTRO_PORTING)
Reason: Add Exherbo distribution option to systemd

From d93e99443c4f6ee65305e254e3b0c3034c2ea73d Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Tue, 13 Jul 2010 11:38:10 -0700
Subject: [PATCH] Add Exherbo distribution

---
 Makefile.am                 |    6 ++++++
 configure.ac                |   11 ++++++++++-
 units/exherbo/font.target   |   12 ++++++++++++
 units/exherbo/font@.service |   18 ++++++++++++++++++
 units/getty@.service.m4     |    1 +
 5 files changed, 47 insertions(+), 1 deletions(-)
 create mode 100644 units/exherbo/font.target
 create mode 100644 units/exherbo/font@.service

diff --git a/Makefile.am b/Makefile.am
index 725685d..f961a9f 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -259,6 +259,12 @@ dist_systemunit_DATA += \
 	units/arch/reboot.service
 endif
 
+if TARGET_EXHERBO
+dist_systemunit_DATA += \
+	units/exherbo/font.target \
+	units/exherbo/font@.service
+endif
+
 dist_doc_DATA = \
 	README \
 	LICENSE \
diff --git a/configure.ac b/configure.ac
index f8a4247..cc56c90 100644
--- a/configure.ac
+++ b/configure.ac
@@ -245,7 +245,7 @@ AM_CONDITIONAL(HAVE_XSLTPROC, test x"$XSLTPROC" != x)
 
 AC_PATH_PROG([M4], [m4])
 
-AC_ARG_WITH(distro, AS_HELP_STRING([--with-distro=DISTRO],[Specify the distribution to target: One of fedora, suse, debian, arch, gentoo, slackware or other]))
+AC_ARG_WITH(distro, AS_HELP_STRING([--with-distro=DISTRO],[Specify the distribution to target: One of fedora, suse, debian, arch, gentoo, slackware, exherbo or other]))
 if test "z$with_distro" = "z"; then
         if test "$cross_compiling" = yes; then
                 AC_MSG_WARN([Target distribution cannot be reliably detected when cross-compiling. You should specify it with --with-distro (see $0 --help for recognized distros)])
@@ -256,6 +256,7 @@ if test "z$with_distro" = "z"; then
                 AC_CHECK_FILE(/etc/arch-release,with_distro="arch")
                 AC_CHECK_FILE(/etc/gentoo-release,with_distro="gentoo")
                 AC_CHECK_FILE(/etc/slackware-version,with_distro="slackware")
+                AC_CHECK_FILE(/etc/exherbo-release,with_distro="exherbo")
         fi
         if test "z$with_distro" = "z"; then
                 with_distro=`uname -s`
@@ -320,6 +321,13 @@ case $with_distro in
                 AC_DEFINE(TARGET_SLACKWARE, [], [Target is Slackware])
                 M4_DISTRO_FLAG=-DTARGET_SLACKWARE=1
                 ;;
+        exherbo)
+                SYSTEM_SYSVINIT_PATH=/etc/init.d
+                SYSTEM_SYSVRCND_PATH=/etc
+                SPECIAL_SYSLOG_SERVICE=syslog-ng.service
+                AC_DEFINE(TARGET_EXHERBO, [], [Target is Exherbo])
+                M4_DISTRO_FLAG=-DTARGET_EXHERBO=1
+                ;;
         other)
                 AS_IF([test "x$with_sysvinit_path" = "x"],
                         [AC_MSG_ERROR([With --distro=other, you must pass --with-sysvinit-path= to configure])])
@@ -362,6 +370,7 @@ AM_CONDITIONAL(TARGET_DEBIAN, test x"$with_distro" = xdebian)
 AM_CONDITIONAL(TARGET_ARCH, test x"$with_distro" = xarch)
 AM_CONDITIONAL(TARGET_GENTOO, test x"$with_distro" = xgentoo)
 AM_CONDITIONAL(TARGET_SLACKWARE, test x"$with_distro" = xslackware)
+AM_CONDITIONAL(TARGET_EXHERBO, test x"$with_distro" = xexherbo)
 
 AC_DEFINE_UNQUOTED(SPECIAL_SYSLOG_SERVICE, ["$SPECIAL_SYSLOG_SERVICE"], [Syslog service name])
 
diff --git a/units/exherbo/font.target b/units/exherbo/font.target
new file mode 100644
index 0000000..22983be
--- /dev/null
+++ b/units/exherbo/font.target
@@ -0,0 +1,12 @@
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Console fonts
+
+[Install]
+WantedBy=multi-user.target
diff --git a/units/exherbo/font@.service b/units/exherbo/font@.service
new file mode 100644
index 0000000..1910f5a
--- /dev/null
+++ b/units/exherbo/font@.service
@@ -0,0 +1,18 @@
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 2 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Console font on %I
+Before=font.target
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+ExecStart=/usr/bin/setfont -C /dev/%I default8x16
+
+[Install]
+Alias=font.target.wants/font@tty1.service font.target.wants/font@tty2.service font.target.wants/font@tty3.service font.target.wants/font@tty4.service font.target.wants/font@tty5.service font.target.wants/font@tty6.service
diff --git a/units/getty@.service.m4 b/units/getty@.service.m4
index 4b65d5b..942b3d6 100644
--- a/units/getty@.service.m4
+++ b/units/getty@.service.m4
@@ -10,6 +10,7 @@ m4_ifdef(`TARGET_SUSE', `m4_define(`GETTY', `/sbin/mingetty')')m4_dnl
 m4_ifdef(`TARGET_DEBIAN', `m4_define(`GETTY', `/sbin/getty 38400')')m4_dnl
 m4_ifdef(`TARGET_GENTOO', `m4_define(`GETTY', `/sbin/agetty 38400')')m4_dnl
 m4_ifdef(`TARGET_ARCH', `m4_define(`GETTY', `/sbin/agetty -8 38400')')m4_dnl
+m4_ifdef(`TARGET_EXHERBO', `m4_define(`GETTY', `/sbin/agetty 38400')')m4_dnl
 m4_dnl
 [Unit]
 Description=Getty on %I
-- 
1.7.2.2

